### PR TITLE
Add e2e test for the AggregateStatus hook

### LINKED ISSUE #1237: [Feature] ResourceInterpreter framework supports `AggregateStatus`
**What would you like to be added**:
Enable [AggregateStatus](https://github.com/karmada-io/karmada/blob/eb4b3e27bc9d080377b899ce251f49ce3f67fbd5/pkg/apis/config/v1alpha1/resourceinterpreterwebhook_types.go#L96-L98) operation for ResourceInterpreter framework.

Tasks:
- [x] Implements default interpreter (#1253)
- [x] Implements customized interpreter (#1253)
- [ ] E2E testing (#1286)

**Why is this needed**:

/assign @iawia002

### PR DESCRIPTION
**Pull request by iawia002**
Signed-off-by: Xinzhao Xu <z2d@jifangcheng.com>

**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Add e2e test case for #1253

**Which issue(s) this PR fixes**:

close #1237 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/cc @XiShanYongYe-Chang @RainbowMango

### CODE REVIEW

File: test/e2e/resourceinterpreter_test.go [hunk 1]
```diff
@@ -178,4 +179,57 @@ var _ = ginkgo.Describe("Resource interpreter webhook testing", func() {
...
+				ClusterNames: framework.ClusterNames(),
+			},
+		})
+
+		ginkgo.It("AggregateStatus testing", func() {
+			framework.CreatePropagationPolicy(karmadaClient, policy)
+			framework.CreateWorkload(dynamicClient, workload)
+
+			ginkgo.By("check whether the workload status can be correctly collected", func() {
+				for _, cluster := range framework.ClusterNames() {
```
Comment (XiShanYongYe-Chang) [inline test/e2e/resourceinterpreter_test.go:206]: Can you add some comments here? This indicates that the operation simulates the member clusters to process workloads.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Pretty good, I think this case can cover the AggregateStatus test for `workload` CRD.

LGTM

File: test/e2e/resourceinterpreter_test.go (hunk 1 above)
Comment (iawia002) [reply test/e2e/resourceinterpreter_test.go:207]: Updated, PTAL.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm

**Review by RainbowMango — APPROVED** [review]
/approve

Seems the testing only tested the `customized` webhook, have you ever thought about covering the default interpreters? (By E2E or unit testing?)

**Comment by iawia002** [comment]
> Seems the testing only tested the customized webhook, have you ever thought about covering the default interpreters? (By E2E or unit testing?)

Yes, the original deployment status collecting test covers the default interpreters case.

**Pull request merged**